### PR TITLE
Use constant time updates to keep 4-band smooth.

### PR DESCRIPTION
**Pull request by cnlohr**

### CODE REVIEW

**Comment by pema99** [comment]
Adding a note to myself here. Lmk if this is ready

File: AudioLink/Shaders/AudioLink.shader [hunk 1]
```diff
@@ -319,8 +319,31 @@ Shader "AudioLink/Internal/AudioLink"
 
                 // If part of the delay
                 } else {
-                    // Return pixel to the left
-                    return AudioLinkGetSelfPixelData(ALPASS_AUDIOLINK + int2(coordinateLocal.x - 1, coordinateLocal.y));
+                    // Slide pixels (coordinateLocal.x > 0)
+                    float4 lastval_timing = AudioLinkGetSelfPixelData(ALPASS_GENERALVU + int2(4, 1)); // Timing for 4-band, move at 90 Hz.
```
Comment (pema99) [inline AudioLink/Shaders/AudioLink.shader:323]: note: naming using snake_case here

File: AudioLink/Shaders/AudioLink.shader [hunk 2]
```diff
@@ -319,8 +319,31 @@ Shader "AudioLink/Internal/AudioLink"
...
+                        {
+                            // For the rest of the line, move by the appropriate speed
+                            return AudioLinkGetSelfPixelData(ALPASS_AUDIOLINK + int2(coordinateLocal.x - frames_to_roll, coordinateLocal.y));
+                        }
+                        else
+                        {
+                            // For the first part, extrapolate the cells.
+                            float last = AudioLinkGetSelfPixelData(ALPASS_AUDIOLINK + int2(0, coordinateLocal.y));
+                            float next = AudioLinkGetSelfPixelData(ALPASS_AUDIOLINK + int2(1, coordinateLocal.y));
+                            float lperv = (coordinateLocal.x - 1) / (float)frames_to_roll;
```
Comment (pema99) [inline AudioLink/Shaders/AudioLink.shader:343]: is 'lperv' a typo here? what is the abbreviation

Comment (cnlohr) [reply AudioLink/Shaders/AudioLink.shader:343]: Prev is correct.  will address.

File: AudioLink/Shaders/AudioLink.shader (hunk 1 above)
Comment (cnlohr) [reply AudioLink/Shaders/AudioLink.shader:323]: Will resolve.

File: AudioLink/Shaders/AudioLink.shader [hunk 3]
```diff
@@ -525,6 +548,17 @@ Shader "AudioLink/Internal/AudioLink"
                             return AudioLinkGetSelfPixelData(ALPASS_CCCOLORS+uint2(1+coordinateLocal.x,0));
                         }
                     }
+                    else if( coordinateLocal.x == 4 )
```
Comment (pema99) [inline AudioLink/Shaders/AudioLink.shader:551]: I'm assuming you made sure this doesn't overlap on the CRT with any of the other newer features? I don't really have a good overview myself

File: AudioLink/Shaders/AudioLink.shader [hunk 4]
```diff
@@ -525,6 +548,17 @@ Shader "AudioLink/Internal/AudioLink"
...
+                    {
+                        // Computation for history timing.
+                        float4 lastval = AudioLinkGetSelfPixelData(ALPASS_GENERALVU + int2(4, 1)); // Timing for 4-band, move at 90 Hz.
+                        lastval.x += unity_DeltaTime.x * AUDIOLINK_4BAND_TARGET_RATE;
+                        // This looks like a frac() but I want to make sure the math gets done the same here
+                        // to prevent any possible mismatch between here and the use of finding the int.
+                        int frames_to_roll = floor( lastval.x );
```
Comment (pema99) [inline AudioLink/Shaders/AudioLink.shader:558]: naming here too :)

File: AudioLink/Shaders/AudioLink.shader (hunk 3 above)
Comment (cnlohr) [reply AudioLink/Shaders/AudioLink.shader:551]: I will need to triple check - but that's what the dev branch is for!

File: AudioLink/Shaders/AudioLink.shader (hunk 4 above)
Comment (pema99) [reply AudioLink/Shaders/AudioLink.shader:558]: you missed this one, charles

**Comment by cnlohr** [comment]
fixed

**Comment by pema99** [comment]
lgtm

**Pull request merged**